processing/src/main/java/org/apache/carbondata/processing/store/CarbonFactDataHandlerColumnar.java

```diff
@@ -195,7 +195,7 @@ public void initialise() throws CarbonDataWriterException {
    * @param row
    * @throws CarbonDataWriterException
    */
-  public void addDataToStore(CarbonRow row) throws CarbonDataWriterException {
+  public synchronized void addDataToStore(CarbonRow row) throws CarbonDataWriterException {
```

Comment:
I think "synchronized" will decrease the performance while many threads call "addDataToStore",  can you check this scene ?

Reply:
@qiuchenjian After changing "addaDataToStore" as synchronized, performance is increased. Earlier it was taking 2.77 hrs for loading 3.5 billion records. After the fix, it is taking 1.87 hrs.

Reply:
@shivamasn the other scenes of  local_sort and global_sort, do you revert the synchronized?

Reply:
@qiuchenjian yes...we are synchronizing only in case of no sort.